docs/cli/ilab-config-current-state.md

```diff
@@ -0,0 +1,90 @@
+# `ilab` Configuration Design as of 0.18.0
+
+## Config Terminology
+
+### Config Layers
+
+There are several locations where config can be specified. These are defined below in order of ascending precedence:
+
+1. The `Config` class in `configuration.py` stores all the general config defaults in code
+1. `config.yaml` is where the user defines their own set of defaults for the system. This can be generated using `ilab config init` and modified using `ilab config edit`. Users can also provide partial config files, for which the rest of the defaults will resort to those defined in the `Config` class.
+1. The user can specify CLI flags when running an `ilab` subcommand, which will override all other configuration.
+
+### Click-isms
+
+[Click](https://click.palletsprojects.com/en/8.1.x/) is a Python library that provides various utilities for Command-Line Interfaces. `ilab` uses these to varying extents to store or pass around config state:
```

Comment:
This links to docs for 8.1.x; the main URL https://click.palletsprojects.com/ looks like it redirects to the latest version. 

I don't know how we manage dependency upgrades and so to what extent this is a tradeoff between linking to outdated vs too current documentation.